napari/utils/events/event.py

```diff
@@ -890,20 +890,23 @@ class WarningEmitter(EventEmitter):
 
     def __init__(
         self,
-        message,
-        category=FutureWarning,
-        stacklevel=3,
+        message: str,
+        category: Type[Warning] = FutureWarning,
+        stacklevel: int = 3,
+        warn_on_connect: bool = True,
         *args,
         **kwargs,
     ) -> None:
         self._message = message
         self._warned = False
         self._category = category
         self._stacklevel = stacklevel
+        self._warn_on_connect = warn_on_connect
         EventEmitter.__init__(self, *args, **kwargs)
 
     def connect(self, cb, *args, **kwargs):
-        self._warn(cb)
+        if self._warn_on_connect:
+            self._warn(cb)
```

Comment:
Why add this? I prefer people to update code, not silence warning.

Reply:
It was emitting a warning when instantiating the event. It should only emit when used.

Reply:
removed, thanks @Czaki